types/bitcore-lib/index.d.ts

```diff
@@ -292,7 +292,44 @@ export class Address {
     readonly network: Networks.Network;
     readonly type: string;
 
-    constructor(data: Buffer | Uint8Array | string | object, network?: Networks.Network, type?: string);
+    constructor(
+        data: Buffer | Uint8Array | string | object,
+        network?: string | Networks.Network,
+        type?: string,
+        multisigType?: string,
+    );
+
+    static createMultisig(
+        publicKeys: PublicKey[],
+        threshold: number,
+        network: string | Networks.Network,
+        nestedWitness?: boolean,
+        type?: string,
+    ): Address;
+    static payingTo(script: Script, network: string | Networks.Network): Address;
+
+    static fromPublicKey(data: PublicKey, network: string | Networks.Network): Address;
+    static fromPublicKeyHash(hash: Buffer, network: string | Networks.Network): Address;
+    static fromScriptHash(hash: Buffer, network: string | Networks.Network): Address;
+    static fromScript(script: Script, network: string | Networks.Network): Address;
+    static fromBuffer(buffer: Buffer, network?: string | Networks.Network, type?: string): Address;
+    static fromString(str: string, network?: string | Networks.Network, type?: string): Address;
+    static fromObject(obj: object): Address;
+
+    static getValidationError(data: string, network?: string | Networks.Network, type?: string): null | Error;
+
+    toBuffer(): Buffer;
+    toObject(): object;
```

Comment:
This return type can probably be more specific: <https://github.com/bitpay/bitcore/blob/bed0702ff56ca207a09711fd2e2a2b75374b51ac/packages/bitcore-lib/lib/address.js#L615-L619>

```suggestion
    toObject(): {
        hash: string;
        type: string;
        network: string;
    };
```